dateutil/parser/_parser.py

```diff
@@ -57,7 +57,7 @@ class _timelex(object):
     # Fractional seconds are sometimes split by a comma
     _split_decimal = re.compile("([.,])")
 
-    def __init__(self, instream):
+    def __init__(self, instream, join_numeric_separated_by_comma=True):
```

Comment:
Hm.. I don't love this. `_timelex` is semi-public, and this is not something I would ever put in a public API.

Reply:
I agree. But we need either to select one behavior or implement some flag in this fashion or completely separate de dependencies. What is your preference?

Reply:
Although I don't like this design I put this so (1) you can identify easily what the problem is and (2) we can discuss better a possible implementation of the fix.

Reply:
Although I don't like this design I put this so (1) you can identify easily what the problem is and (2) we can discuss better a possible implementation of the fix.

Reply:
Hmmm... I think I see what's going on here.

I find it a bit strange that `_tzparser` re-uses the lexer like this. I see three other possible ways of going here:

1. Split out `_timelexer` and `_tzlexer` into two different lexers with independent rules.
2. Do some sort of token splitting downstream based on the context.
3. Reverse  the commit that keeps the comma token together and dynamically re-join it as necessary.

None of them are particularly satisfying.

Another way to go with your approach is to come up with some more satisfying name for that flag - if there's some unifying theme for the cases where you'd want it to be true, then I'm more OK with adding the flag.

The thing is, 2 and 3 seem wrong because they violate separation of concerns, but we may need to do either 2 or 3 if we want to also solve #203 and other comma-related problems caused by this.

Reply:
If we want to move meanwhile we implement 1,2 or 3, we can use a new private API to `_timelex` that activates the flag so this is invisible to the end user. Is not good, but is lightweight and it will keep this PR logically separated from bigger fixes. I think 3 underscores will be enough for privacy (`_parser._timelex._privatestuff`). :wink:

Reply:
Having gone through the comments but not yet the code: is there a viable solution here that does not require addressing the `_timelex` design?  If so, I expect that is the best short-term solution.

Reply:
This touches on #541, and as @pganssle says ideally that design question should be addressed separately.

IIRC the comma-as-decimal thing is mostly there for the python logging format.  If that is correct, we can more strictly check for something like `'(?P<h>\d{1,2}):(?P<m>\d{2})(:(?P<s>\d{2}([,\.]\d+)?))` (though likely not using regex)

Reply:
@jbrockmendel There are a large number of possible hacky workarounds we could do, but it's basically the same problem as with [this PR](https://github.com/dateutil/dateutil/pull/279#discussion_r71794602) - right now we have some tokens that are combined when they shouldn't be, so the options are:

1. Add an additional tokenizer step in the parser that splits these aggressively-combined tokens (no direct effect on `_timelex`, but does couple to the current behavior)
2. Make it so `_timelex` doesn't combine these anymore and add a re-combination step in the parser portion.
3. Some variation on Pablo's solution, e.g. "tell `_timelex` which behavior we want when we call it" - this can either be a flag like what Pablo is doing or we subclass `_timelex` to make a `_tzlex`, and the two vary depending on this behavior.

I don't see any other obvious solution, since in some contexts you want the tokens combined and in some contexts you want the tokens split.